lib/puppet/provider/ibm_pkg/imcl.rb

```diff
@@ -77,13 +89,19 @@ def self.find_installed_xml(user)
 
     installed_xml_path = nil
 
-    user_path = (user == 'root') ? '/var/ibm/' : "/home/#{user}/var/ibm/"
+    user_path = (user == 'root') ? '/var/ibm/' : "#{user_home(user)}/var/ibm/"
 
     if File.exist? user_path
       Find.find(user_path) { |path| installed_xml_path = path if path =~ %r{InstallationManager/installed.xml$} }
     end
 
-    installed_xml_path if File.file?(installed_xml_path)
+    begin
+      if File.file?(installed_xml_path)
+        return installed_xml_path
+      end
+    rescue RuntimeError => e
```

Comment:
What raises a `RuntimeError`? `File.file?` just returns `false` for absent files.

Reply:
Oh, `File.file?(nil)` does, but it's bad practice to just pass nil and catch the error. The code should check if `installed_xml_path` is nil instead.